src/deploy.ts

```diff
@@ -83,6 +83,34 @@ export class HypERC20Deployer extends GasRouterDeployer<
     }
   }
 
+  // Gets the metadata for a collateral token, favoring the config
+  // and getting any on-chain metadata that is missing.
+  async getCollateralMetadata(
+    chain: ChainName,
+    config: CollateralConfig,
+  ): Promise<ERC20Metadata> {
+    const metadata = {
+      name: config.name,
+      symbol: config.symbol,
+      decimals: config.decimals,
+      totalSupply: 0,
+    };
+
+    if (metadata.name !== undefined && metadata.symbol !== undefined && metadata.decimals !== undefined) {
```

Comment:
Check for null to:
`if (metadata.name && metadata.symbol && metadata.decimals !== undefined && metadata.decimals !== null)`